pages/orders/[orderId]/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, Flex, H1, Panel, StatefulTable } from '@bigcommerce/big-design';
+import { useRouter } from 'next/router';
+import React from 'react';
+import styled from 'styled-components';
+
+import ErrorMessage from '../../../components/error';
+import Loading from '../../../components/loading';
+import { useShippingAndProductsOrder } from '../../../lib/hooks';
+import { BillingAddress, OrderProduct, ShippingAndProductsOrder } from '../../../types';
+
+const StyledAddress = styled.address`
+    color: ${({ theme }) => theme.colors.secondary70};
+    font-style: normal;
+    line-height: ${({ theme }) => theme.lineHeight.small};
+`;
+
+const StyledUl = styled.ul`
+    ${({ theme }) => theme.helpers.listReset}
+
+    color: ${({ theme }) => theme.colors.secondary70};
+    line-height: ${({ theme }) => theme.lineHeight.xxLarge};
+`;
+
+const InternalOrderPage = (order: ShippingAndProductsOrder) => {
```

Comment:
a part of me wonders if we should move this functional component to its own component page. Traditionally, with class based components, we'd have only one component per page. With hooks/ FC, that line has become a bit blurry insomuch that might only do it if the multiple components contain state and/ or side effects. 

What makes me think about it here is that this component itself is rendering multiple smaller components and is being called by `OrderPage` which itself has state that's managed by SWR. FWIW, I believe we're doing something similar in `products/[pid]`.

Reply:
The idea here is that we are requiring that `order` be populated before we render the page. It's a better practice to ensure that you have data passed into your components rather than allowing `Type | undefined` being passed in. This helps keeps unnecessary checks outside of our components. We have been taking this approach at BigCommerce within our NextJS applications. Ask me offline for some examples of this.

It's not necessarily bad to have multiple components in page, especially if one is a simple one for data fetching and one for rending. If we wanted to move to a single component per file approach it's a lot of lift since every file/folder in `/pages` in a NextJS application is a route. We have solutions in place in our internal applications, but we enforce the patterns via custom linting rules and TypeScript module resolution.